### PR TITLE
#167165046 Setup postgres database

### PR DESCRIPTION
**Pull request by Tunji17**
### What does this PR do?
- Changes the database from mongo to Postgres
#### Description of Task to be completed?
- Used sequelize ORM to interface with a Postgres database for both test and development
#### How should this be manually tested?

- clone this repo using git clone 
- and run `npm run migrate`

#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[#167165046](https://www.pivotaltracker.com/story/show/167165046)

### CODE REVIEW

**Review by tolulope-od — APPROVED** [review]

File: database/models/index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  .readdirSync(__dirname)
+  .filter(file => {
+    return (file.indexOf('.') !== 0) && (file !== basename) && (file.slice(-3) === '.js');
+  })
+  .forEach(file => {
+    const model = sequelize['import'](path.join(__dirname, file));
+    db[model.name] = model;
+  });
+
+Object.keys(db).forEach(modelName => {
```
[the remark this answers is not in the document]
Comment (mazma1) [reply database/models/index.js]: Fixing this won't break anything

File: database/models/index.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+} else {
+  sequelize = new Sequelize(config.database, config.username, config.password, config);
+}
+
+fs
+  .readdirSync(__dirname)
+  .filter(file => {
+    return (file.indexOf('.') !== 0) && (file !== basename) && (file.slice(-3) === '.js');
+  })
+  .forEach(file => {
```
[the remark this answers is not in the document]
Comment (mazma1) [reply database/models/index.js]: Same here, this won't break anything

File: database/models/index.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+let sequelize;
+if (config.use_env_variable) {
+  sequelize = new Sequelize(process.env[config.use_env_variable], config);
+} else {
+  sequelize = new Sequelize(config.database, config.username, config.password, config);
+}
+
+fs
+  .readdirSync(__dirname)
+  .filter(file => {
```
[the remark this answers is not in the document]
Comment (mazma1) [reply database/models/index.js]: Fix this as well

File: database/models/index.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const Sequelize = require('sequelize');
```
[the remark this answers is not in the document]
Comment (mazma1) [reply database/models/index.js]: Fix this

File: database/models/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tunji17) [reply database/models/index.js]: okay let me work on it

File: database/models/index.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+'use strict';
```
[the remark this answers is not in the document]
Comment (mazma1) [reply database/models/index.js]: Removing 'use strict' wont break anything

File: database/config/config.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  use_env_variable: 'DATABASE_URL',
+};
+
+export const test = {
+  use_env_variable: 'TEST_DATABASE_URL'
+};
+
+export const production = {
+  use_env_variable: 'DATABASE_URL'
+};
```
[the remark this answers is not in the document]
Comment (mazma1) [reply database/config/config.js]: Every file should end with an empty line so fix this

File: database/models/like.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    articleId: {
+      type: DataTypes.INTEGER
+    },
+    commentId: {
+      type: DataTypes.INTEGER
+    },
+    likedOn: {
+      allowNull: false,
+      type: DataTypes.DATE,
+      defaultValue: DataTypes.literal('CURRENT_TIMESTAMP'),
```
Comment (tolulope-od) [inline database/models/like.js]: Please delete this. It shouldn't be in the model definition. It's okay in the migrations

File: database/models/article.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    },
+  }, {});
+  Article.associate = function(models) {
+    Article.hasMany(models.ArticleImage, {
+      foreignKey: 'articleId',
+      as: 'articleimage',
+      onDelete: 'CASCADE',
+      onUpdate: 'CASCADE',
+    });
+    Article.hasMany(models.Bookmarks, {
```
Comment (tolulope-od) [inline database/models/article.js]: This should be `Bookmark`, not `Bookmarks`

File: database/models/User.js [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+      foreignKey: 'userId',
+      as: 'userreport',
+      onDelete: 'CASCADE',
+      onUpdate: 'CASCADE',
+    });
+    User.belongsToMany(models.User, {
+      foreignKey: 'userId',
+      otherKey: 'followerId',
+      as: 'followers',
+      through: 'followers',
```
Comment (tolulope-od) [inline database/models/User.js]: Please change the `followers` to `Followers` so it references the table properly. Thank you

File: database/models/User.js [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+      timestamps: false,
+    });
+    User.belongsToMany(models.User, {
+      foreignKey: 'followerId',
+      otherKey: 'userId',
+      as: 'followings',
+      through: 'followers',
```
Comment (tolulope-od) [inline database/models/User.js]: The same goes for this line; `followers` to `Followers`. Thanks

File: database/models/User.js [hunk 11]
```diff
@@ -0,0 +1,95 @@
+/* eslint-disable space-before-function-paren */
```
Comment (tolulope-od) [inline database/models/User.js:1]: Please change the name of this file, it should start with a lowercase since it doesn't export a class. Thank you

**Review by tolulope-od — COMMENTED** [review]
Great job Tunji! You've made some colossal changes to the codebase, please see my comments on some issues I found and kindly implement the fixes I proposed.

**Pull request closed without merging**